### DISCUSSION TITLE
Select line from first non whitespace char to end

### DISCUSSION THREAD

**Discussion by CptPotato** [helix-editor/helix#2503] [Q&A] [answered]
I'm looking for some way to extend the selection to the line bounds (like the current <kbd>x</kbd> / <kbd>X</kbd>) but *starting at the first non-whitespace character*.

Is it currently possible to bind a key that does that? I'm trying to get a similar behavior to vim's <kbd>Shift</kbd> <kbd>C</kbd>.

**Comment by the-mikedavis** [comment, marked as answer]
This might not work quite the same but you can extend to line bounds with <kbd>X</kbd> and then trim the whitespace from the ends of the selection with <kbd>_</kbd>

**Reply by CptPotato** [reply]
Nice, I didn't know about <kbd>_</kbd>.

<kbd>x</kbd><kbd>_</kbd><kbd>c</kbd> is a whole lot better than <kbd>g</kbd><kbd>s</kbd><kbd>v</kbd><kbd>g</kbd><kbd>l</kbd><kbd>c</kbd> :eyes:

I also bound the selection part to <kbd>Alt</kbd><kbd>x</kbd> for convinience: `"A-x" = [ "extend_to_line_bounds", "trim_selections" ]
`

**Comment by peter9477** [comment]
In case trimming the trailing whitespace wasn't desirable in this case, `g` `s` `t` `Enter` can *almost* be used instead, selecting up to the newline, but it does select the following line if used on a completely empty line or, strangely, one with only a single character.  I suspect the latter is a bug.

**Reply by Omnikar** [reply]
If you're on a line with a single character, then the cursor is already positioned just before the newline character, so going to the next one on the next line is intended behaviour.

**Reply by peter9477** [reply]
As it relates to this issue, you may be right, but I believe nevertheless there's a bug in that functionality, as reported here now: https://github.com/helix-editor/helix/issues/4065
